### PR TITLE
Remove unused `NOTIFICATION_NODE_RECACHE_REQUESTED` notification

### PR DESCRIPTION
**Pull request by YuriSizov**
This is a follow-up to https://github.com/godotengine/godot/pull/82471. The notification was never used by the engine since the moment it was introduced in Godot 4 during its closing days of the development (https://github.com/godotengine/godot/pull/57606). It was sent, but never received by anything. While https://github.com/godotengine/godot/pull/82471 attempted to finally utilize it, the consensus was that the notification was a mistake and we should resolve the issue another way. Which we did.

So now the notification can be removed. This breaks compatibility, but there is no reason to assume users would rely on it in practice.

It also conflicts with some other notifications (see https://github.com/godotengine/godot/issues/75839), making it a problem for some GDExtensions.

### CODE REVIEW

**Review by fire — APPROVED** [review]
Looks good from me. I was part of the discussions about the re-caching bugs in the past.

**Review by SaracenOne — APPROVED** [review]
👍

**Review by raulsntos — APPROVED** [review]
This change doesn't break binary compatibility in C#, only source compatibility, because it removes a constant field.

File: misc/extension_api_validation/4.1-stable.expected [hunk 1]
```diff
@@ -305,3 +305,10 @@ Validate extension JSON: Error: Field 'classes/LineEdit/properties/right_icon':
 Validate extension JSON: Error: Field 'classes/Sprite3D/properties/texture': type changed value in new API, from "Texture" to "Texture2D".
 
 Changed the hint types of some nodes' properties from Texture to Texture2D. No adjustments should be necessary.
+
+
+GH-84419
+--------
+Validate extension JSON: API was removed: classes/Node/constants/NOTIFICATION_NODE_RECACHE_REQUESTED
+
+Remove unused NOTIFICATION_NODE_RECACHE_REQUESTED notification
```
Comment (Bromeon) [inline misc/extension_api_validation/4.1-stable.expected:314]: Maybe additionally mention that it actively caused a conflict? (which is imo the better reason to break compat than being unused)

Comment (YuriSizov) [reply misc/extension_api_validation/4.1-stable.expected:314]: Are these comments used anywhere? There is already a link to the PR which explains the context.

**Review by akien-mga — APPROVED** [review]
Looks good.

File: misc/extension_api_validation/4.1-stable.expected (hunk 1 above)
Comment (Bromeon) [reply misc/extension_api_validation/4.1-stable.expected]: Not sure, but I guess if they weren't, we could just post the link to the PR without description.

Anyway, it's a minor thing, thanks for the change! 👍

**Pull request merged**